app/controllers/search_controller.rb

```diff
@@ -11,4 +11,14 @@ def extract_filters
     respond_with TextSearchClient.parse_text(params[:q])
   end
 
+  def log_metrics_event
+    if params[:type] && params[:data] then
+      if params[:other_data] && params[:other_data] != "" then
+        metrics_event(params[:type], params[:data], eval(params[:other_data])) 
+      else 
+        metrics_event(params[:type], params[:data])
+      end
+    end
+    render nothing: true, status: :ok, content_type: "text/html"
```

Comment:
```respond_with :nothing``` should be good enough. And content_type is not needed for ':nothing'.

Reply:
It wasn't good enough - it's the first thing I tried.  :\  It throws this error:

NoMethodError (undefined method `nothing_url' for #<SearchController:0x007ffa413f44e8>):
  app/controllers/search_controller.rb:22:in `log_metrics_event'

It put me on the right track, though; it led me to the syntax you see here.  The syntax above, works - no exceptions and behaves.

Reply:
You sure it is ```respond_with :nothing```, not ```render :nothing```?

Reply:
Yes.  I tried it multiple times.  The code I have here works; the respond_with :nothing does not.

Reply:
I'll add - I'm aware that "respond_with :nothing" *should be* acceptable - I really don't know why it's throwing an exception.  However, the line used above absolutely does work and my feeling is that is what is important.